runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/DokkaCollectorTask.kt

```diff
@@ -0,0 +1,55 @@
+package org.jetbrains.dokka.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.UnknownTaskException
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import java.lang.IllegalStateException
+
+open class DokkaCollectorTask : DefaultTask() {
+
+    @Input
+    var modules: List<String> = emptyList()
+
+    @Input
+    var outputDirectory: String = ""
+
+    private lateinit var configuration: GradleDokkaConfigurationImpl
+
+    @TaskAction
+    fun collect() {
+        val passesConfigurations = getProjects(project).filter { it.name in modules }.map {
+            val task = try {
+                it.tasks.getByName(DOKKA_TASK_NAME) as DokkaTask
```

Comment:
if you add `compileOnly(gradleKotlinDsl())` dependency you can use typesafe variants here.